pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -932,8 +935,52 @@ private void increaseAvailablePermits(ClientCnx currentCnx, int delta) {
         }
     }
 
+    private ByteBuf decryptPayloadIfNeeded(MessageIdData messageId, MessageMetadata msgMetadata, ByteBuf payload,
+            ClientCnx currentCnx) {
+
+        if (msgMetadata.getEncryptionKeysCount() == 0) {
+            return payload.retain();
+        }
+
+        // If KeyReader is not configured throw exception based on config param
+        if (conf.getCryptoKeyReader() == null) {
+
+            if (conf.getCryptoFailureAction() == ConsumerCryptoFailureAction.CONSUME) {
+                log.warn("CryptoKeyReader interface is not implemented. Consuming encrypted message.");
```

Comment:
can we also log `subscription` and `consumerName` in msg?

Reply:
Fixed.